### PR TITLE
Use updated config to specify PubSub channel for Brokers

### LINKED ISSUE #606: Use the updated way of specifying a Broker's default Channel
**Describe the bug**
Knative has updated how Broker determine the default channel to use https://github.com/knative/eventing/pull/2651

The current way of updating default-ch-webhook doesn't work anymore, so all newer Brokers will use an InMemory channel instead of PubSub.

 **Expected behavior**
The Broker's default Channel should be PubSub

 **To Reproduce**
Following the steps in google/knative-gcp and Brokers have an InMemory Channel.

https://github.com/google/knative-gcp/blob/master/docs/install/install-broker-with-pubsub-channel.md

### PR DESCRIPTION
**Pull request by zargarpur**
Fixes #606

## Proposed Changes

- Add ConfigMap to default to PubSub Channels for Brokers
- Update doc instructions to patch in ConfigMap reference

**Release Note**

```release-note
Knative has updated the way Brokers determine which default Channel to use. Adding a ConfigMap that includes PubSub as the default. 

action required Follow updated instructions to patch Knative's ConfigMap to point use this configuration.
```

### CODE REVIEW

File: config/default-broker-channel.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config-br-default-channel
+  namespace: cloud-run-events
```
Comment (zargarpur) [inline config/default-broker-channel.yaml]: I named this the same as in knative/eventing "config-br-default-channel", but within the cloud-run-events namespace 

https://github.com/knative/eventing/blob/master/config/core/configmaps/default-broker.yaml

File: docs/install/patch-config-br-defaults-with-pubsub.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+# limitations under the License.
+
+data:
+  # Configuration for defaulting channels that do not specify CRD implementations.
+  default-br-config: |
+    clusterDefault:
+      apiVersion: v1
+      kind: ConfigMap
+      name: config-br-default-channel
+      namespace: cloud-run-events
```
Comment (nachocano) [inline docs/install/patch-config-br-defaults-with-pubsub.yaml]: add empty line

Comment (vaikas) [inline docs/install/patch-config-br-defaults-with-pubsub.yaml]: nit, add newline?

File: config/default-broker-channel.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2020 The Knative Authors
```
Comment (nachocano) [inline config/default-broker-channel.yaml]: I re-arranged the config folders a bit to match eventing and serving... 
this should go inside config/core/configmaps. Let's rename it to be `config-br-default-channel.yaml`

And then you create a symlink to it from config. Call it `400-config-br-default-channel.yaml`

File: docs/install/patch-config-br-defaults-with-pubsub.yaml (hunk 2 above)
Comment (nachocano) [reply docs/install/patch-config-br-defaults-with-pubsub.yaml]: @vaikas quick question, config-br-defaults in knative-eventing can point to configMaps in other namespaces?

File: config/default-broker-channel.yaml (hunk 3 above)
Comment (zargarpur) [reply config/default-broker-channel.yaml]: Done

**Comment by nachocano** [comment]
Thanks for the quick fix @zargarpur!!!

File: docs/install/patch-config-br-defaults-with-pubsub.yaml (hunk 2 above)
Comment (vaikas) [reply docs/install/patch-config-br-defaults-with-pubsub.yaml]: yes.

Comment (nachocano) [reply docs/install/patch-config-br-defaults-with-pubsub.yaml]: Thanks Ville!

**Pull request merged**